src/sagas.js

```diff
@@ -0,0 +1,24 @@
+import {all, call, put, takeEvery} from 'redux-saga/effects'
+import createSagaMiddleware from 'redux-saga';
+import {rootReducer} from 'reducers/rootReducer';
+import {applyMiddleware, createStore} from 'redux';
+import {watchSpinner} from 'components/Spinner/SpinnerSagas';
+import {signIn} from 'components/App/AppSagas';
+
+const sagaMiddleware = createSagaMiddleware();
+
+export const store = createStore(
+  rootReducer,
+  applyMiddleware(sagaMiddleware)
+);
+
+sagaMiddleware.run(rootSaga);
+
+export const action = (type, data = null) => store.dispatch({type, data});
+
+export function* rootSaga() {
+  yield all([
+    signIn(),
+    watchSpinner()
```

Comment:
Will we need to initilize all watchers here, for each action like showing a spinner?

Reply:
Yes, this is the place where we subscribe saga for action. Not for every action but for every saga file. I will change that naming.